pkg/tap/protocols.go

```diff
@@ -46,7 +53,11 @@ func (s *qemuProtocol) Buf() []byte {
 }
 
 func (s *qemuProtocol) Write(buf []byte, size int) {
-	binary.BigEndian.PutUint32(buf, uint32(size))
+	if size > math.MaxUint32 {
+		log.Warnf("size exceeds max limit. Resetting to: %d", math.MaxInt32)
+		size = math.MaxUint32
+	}
+	binary.BigEndian.PutUint32(buf, uint32(size)) //#nosec: G115. Safely checked
```

Comment:
slightly worried about adding too much code in `Write`, `txBuf`, … as this could have a performance impact (but I haven’t measured it, it’s likely to not be impactful).
Haven’t given much thought to this part of the PR, so I don’t know if we can handle this differently.

Reply:
We could change the signature of `Write` method to  `Write(buf []byte, size uint32)`. This would mean we would only need to check in `hyperkitProtocol.Write` for `uint16`

Reply:
I  suspect these lines https://github.com/containers/gvisor-tap-vsock/blob/f341afd0ebb7700cf421c67b0ef76c3dbf599cba/pkg/tap/switch.go#L169-L170 would then need a check so that we can cast to `uint32` before calling into the function?

Reply:
yes, that's right. We can perform conversion here and then call the Write function. We can return an error in case of an invalid value. However, there's still `hyperkitProtocol` which needs `uint16`.  We could also skip Write in case of an invalid size (but I think this is not correct):

```
        if size > math.MaxUint32 {
		log.Errorf("size exceeds max limit.")
	}  else {
	        binary.BigEndian.PutUint32(buf, uint32(size))
        }
```